### PR TITLE
Developer

### PR DESCRIPTION
**Pull request by AnastasiaOsinska**

### CODE REVIEW

**Comment by vadimux** [comment]
branch should be named `dev` or `development` not `Developer` and `main` branch is `master` as well

File: Podfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Uncomment the next line to define a global platform for your project
+# platform :ios, '9.0'
+
+target 'TodoApp' do
+  # Comment the next line if you don't want to use dynamic frameworks
+  use_frameworks!
+
+  # Pods for TodoApp
+pod 'SideMenu'
```
Comment (vadimux) [inline Podfile:9]: please add an indent

File: Podfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Uncomment the next line to define a global platform for your project
+# platform :ios, '9.0'
```
Comment (vadimux) [inline Podfile:2]: usually specifies the minimum version for pods

File: Podfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Uncomment the next line to define a global platform for your project
```
Comment (vadimux) [inline Podfile:1]: first line usually is `source 'https://github.com/CocoaPods/Specs.git'`

File: Resources/AppDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import UIKit
+
+@main
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+    var window: UIWindow?
+    
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+//        window = UIWindow()
```
Comment (vadimux) [inline Resources/AppDelegate.swift:16]: commented lines should not go into the commit

File: TodoApp/Controllers/ContainerController.swift [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+//  ViewController.swift
+//  TodoApp
+//
+//  Created by Anastasiya Osinskaya on 11/24/20.
+//
+
+import UIKit
+import SideMenu
+
+class ContainerController: UIViewController, MenuControllerDelegate {
```
Comment (vadimux) [inline TodoApp/Controllers/ContainerController.swift:11]: `MenuControllerDelegate` should be as extension

File: TodoApp/Controllers/ContainerController.swift [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+            titleLabel.font = UIFont.boldSystemFont(ofSize: 30)
+            navigationItem.titleView = titleLabel
+        }
+    }
+}
+
+extension ContainerController: UITableViewDelegate, UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        1
```
Comment (vadimux) [inline TodoApp/Controllers/ContainerController.swift:95]: add `return` - it's in the style of methods that return something

File: TodoApp/Controllers/NewTaskController.swift [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        static let boldSystemFont: CGFloat = 30
+    }
+    
+    @IBOutlet weak var taskTitleTextField: UITextField!
+    @IBOutlet weak var saveButton: UIBarButtonItem!
+    @IBOutlet weak var descriptionTextView: UITextView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let titleLabel = UILabel(frame: CGRect(x: 0, y: 0, width: view.frame.width - 10, height: view.frame.height))
```
Comment (vadimux) [inline TodoApp/Controllers/NewTaskController.swift:26]: setting of UI - better put into a separate method

File: TodoApp/Controllers/NewTaskController.swift [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    
+    func textViewDidBeginEditing(_ textView: UITextView) {
+        if descriptionTextView.textColor == UIColor.lightGray {
+            descriptionTextView.text = ""
+            descriptionTextView.textColor = UIColor.black
+        }
+    }
+    
+    func textViewDidEndEditing(_ textView: UITextView) {
+
```
Comment (vadimux) [inline TodoApp/Controllers/NewTaskController.swift:48]: is an empty string needed here if the previous method doesn't have it?

File: TodoApp/Controllers/SettingsViewController/SettingsViewController.swift [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+//  SettingsViewController.swift
+//  TodoApp
+//
+//  Created by Anastasiya Osinskaya on 11/24/20.
+//
+
+import Foundation
+import UIKit
+
+class SettingsViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
```
Comment (vadimux) [inline TodoApp/Controllers/SettingsViewController/SettingsViewController.swift:11]: `UITableViewDelegate, UITableViewDataSource` as extension

**Comment by vadimux** [comment]
>  Your project does not explicitly specify the CocoaPods master specs repo. Since CDN is now used as the default, you may safely remove it from your repos directory via `pod repo remove master`. To suppress this warning please add `warn_for_unused_master_specs_repo => false` to your Podfile.

you need to remove it as done in VKMarket

**Comment by vadimux** [comment]
can't build the project -

> error: Build input files cannot be found: '/Users/_.com/TodoApp/TodoApp/Extensions/SimpleOver.swift', '/Users/_.com/TodoApp/TodoApp/UIComponents/SideMenu.swift' (in target 'TodoApp' from project 'TodoApp')

File: TodoApp/Controllers/TasksTableViewCell/TasksTableViewCell.swift [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        popupDialog.addAction(deleteAction)
+        popupDialog.addAction(cancelAction)
+        
+        parentViewController?.present(popupDialog, animated: true, completion: nil)
+    }
+    
+}
+
+extension UIView {
+    var parentViewController: UIViewController? {
```
Comment (vadimux) [inline TodoApp/Controllers/TasksTableViewCell/TasksTableViewCell.swift:44]: not sure that we should use this, but need check with app..I'll add the comment in the future about it

**Comment by vadimux** [comment]
as I see there are no files SimpleOver.swift and SideMenu.swift

**Review by vadimux — CHANGES_REQUESTED** [review]

File: Podfile (hunk 2 above)
Comment (AnastasiaOsinska) [reply Podfile:2]: solved

File: Podfile (hunk 3 above)
Comment (AnastasiaOsinska) [reply Podfile:1]: solved

File: Resources/AppDelegate.swift (hunk 4 above)
Comment (AnastasiaOsinska) [reply Resources/AppDelegate.swift:16]: solved

File: TodoApp/Controllers/ContainerController.swift (hunk 5 above)
Comment (AnastasiaOsinska) [reply TodoApp/Controllers/ContainerController.swift:11]: solved

File: TodoApp/Controllers/ContainerController.swift (hunk 6 above)
Comment (AnastasiaOsinska) [reply TodoApp/Controllers/ContainerController.swift:95]: done

File: TodoApp/Controllers/NewTaskController.swift (hunk 8 above)
Comment (AnastasiaOsinska) [reply TodoApp/Controllers/NewTaskController.swift:48]: solved

File: TodoApp/Controllers/SettingsViewController/SettingsViewController.swift (hunk 9 above)
Comment (AnastasiaOsinska) [reply TodoApp/Controllers/SettingsViewController/SettingsViewController.swift:11]: solved

File: TodoApp/Controllers/TasksTableViewCell/TasksTableViewCell.swift (hunk 10 above)
Comment (AnastasiaOsinska) [reply TodoApp/Controllers/TasksTableViewCell/TasksTableViewCell.swift:44]: removed

**Pull request closed without merging**